daemon/src/engine/server/CryptoChallege.h

```diff
@@ -0,0 +1,633 @@
+/*
```

Comment:
I got an idea, how about a struct which holds Data and an error string, which overloads operator >> or something so you can do something like this:
```C++
Encrypt encrypt(input);
if ( encrypt >> Base64Decode >> Decrypt("Aes256", key) >> HexDecode )
    doSomething();
else
    print( encrypt.error );
```